<?xml version="1.0"?>
<launch>
    <arg name="openpose_python" default="/home/leo/rasberry_ws/src/openpose/build/python" />
    <arg name="openpose_models" default="/home/leo/rasberry_ws/src/openpose/models" />
    <arg name="posture_classifier_model" default="/home/leo/rasberry_ws/src/mesapro/config/classifier_model_3D_v2.joblib" />    
    <arg name="config_direct" default="/home/leo/rasberry_ws/src/mesapro/config/" /> 
    <arg name="visual_mode" default="1" />   
    <arg name="thermal_info" default="True" />
    <arg name="image_rotation" default="270" />
    <arg name="operation_mode" default="logistics" />  
    <arg name="n_cameras" default="1" />
    <arg name="openpose_visual" default="True" />    
    
    <include file="$(find mesapro)/launch/bags_data_rgbd_thermal.launch"/>

    <node pkg="mesapro" type="human_safety_system.py" name="hri_safety_system">
        <param name="operation_mode" value="$(arg operation_mode)"/>
    </node>

    <node pkg="mesapro" type="human_detector_rgbd_thermal_old.py" name="hri_camera_detector">
        <param name="openpose_python" value="$(arg openpose_python)"/>
        <param name="openpose_models" value="$(arg openpose_models)"/>
        <param name="posture_classifier_model" value="$(arg posture_classifier_model)"/>
        <param name="thermal_info" type="bool" value="$(arg thermal_info)"/>
        <param name="image_rotation" type="int" value="$(arg image_rotation)"/>
        <param name="n_cameras" type="int" value="$(arg n_cameras)"/>
        <param name="openpose_visual" type="bool" value="$(arg openpose_visual)"/>
    </node>

    <node pkg="mesapro" type="human_perception_system_old.py" name="hri_perception"/>   

    <node pkg="mesapro" type="demo_graph_interf_old.py" name="hri_visualization"> 
	<param name="visual_mode" type="int" value="$(arg visual_mode)"/>
        <param name="config_direct" value="$(arg config_direct)"/>
        <param name="image_rotation" type="int" value="$(arg image_rotation)"/>
        <param name="thermal_info" type="bool" value="$(arg thermal_info)"/>
        <param name="n_cameras" type="int" value="$(arg n_cameras)"/>
    </node>
</launch>


